src/network.rs

```diff
@@ -246,232 +224,286 @@ impl Nectar {
+                        chain: ethereum::Chain::Local {
+                            chain_id: u32::from(self._configured_ethereum_network),
+                            dai_contract_address: token_contract,
+                        },
+                    },
                 };
 
-                if ongoing_trade_with_taker_exists {
-                    tracing::warn!(
-                        "ignoring take order request from taker with ongoing trade, taker: {:?}, order: {}",
-                        taker.peer_id(),
-                        order_id,
-                    );
-                    return;
-                }
-
-                let order = match self.orderbook.get_order(&order_id) {
-                    Some(order) => order,
-                    None => {
-                        tracing::warn!(
-                            "unexpected take order request, taker: {}, order: {}",
-                            taker.peer_id(),
-                            order_id,
-                        );
-                        return;
+                let (role_dependant_params, common_params, swap_protocol) = match swap_protocol {
+                    SwapProtocol::HbitHerc20 {
+                        hbit_expiry_offset,
+                        herc20_expiry_offset,
+                    } => {
+                        // todo: do checked addition
+                        #[allow(clippy::cast_sign_loss)]
+                        #[allow(clippy::cast_possible_truncation)]
+                        let (ethereum_absolute_expiry, bitcoin_absolute_expiry) = {
+                            let ethereum = match_reference_point
+                                .add(herc20_expiry_offset.0)
+                                .timestamp() as u32;
+                            let bitcoin =
+                                match_reference_point.add(hbit_expiry_offset.0).timestamp() as u32;
+
+                            (ethereum, bitcoin)
+                        };
+                        match our_position {
+                            Position::Buy => (
+                                RoleDependentParams::Bob(BobParams {
+                                    bitcoin_identity,
+                                    ethereum_identity,
+                                }),
+                                CommonParams {
+                                    erc20: comit::asset::Erc20 {
+                                        token_contract,
+                                        quantity: erc20_quantity,
+                                    },
+                                    bitcoin: quantity,
+                                    ethereum_absolute_expiry,
+                                    bitcoin_absolute_expiry,
+                                    ethereum_chain_id: self._configured_ethereum_network,
+                                    bitcoin_network: self._configured_bitcoin_network,
+                                },
```

Comment:
CommonParams are exactly the same in both cases no? In fact, I think they should be the same for all combinations given that the params themselves don't say anything about the swap protocol.